lib/Handler/AbstractHandler.php

```diff
@@ -30,9 +33,20 @@
      *
      * @param string  $minimalLevel the minimal severity of the LogLevel to start logging with
      * @param integer $verbose      the Verbosity of the Log
+     * @param array   $additionalParameters optional additional parameters required for the Handler
+     * @param \Productsup\Flexilog\Processor $processor the Processor to be used for the Handler, if none supplied
+     *                              the DefaultProcessor one will be used.
      */
-    public function __construct($minimalLevel = 'debug', $verbose = 0)
+    public function __construct($minimalLevel = 'debug',
+                                $verbose = 0,
+                                array $additionalParameters = array(),
+                                ProcessorInterface $processor = null)
     {
+        if (is_null($processor)) {
+            $this->processor = new \Productsup\Flexilog\Processor\DefaultProcessor();
+        } else {
+            $this->processor = $processor;
```

Comment:
```php
$this->processor = $processor ?? new \Productsup\Flexilog\Processor\DefaultProcessor;
```

Reply:
That would make the entire codebase PHP7 minimum, since this feature didn't exist before. http://php.net/manual/en/migration70.new-features.php#migration70.new-features.null-coalesce-op

Currently the codebase is PHP5.6 compatible.